tests/topotests/bgp_snmp_bgp4v2mib/test_bgp_snmp_bgp4v2mib.py

```diff
@@ -220,6 +230,82 @@ def _snmpwalk_med():
+            ("1.3.6.1.2.1.15.3.1.14.192.168.12.1", '"06 04 "'),
+            ("1.3.6.1.2.1.15.3.1.2.192.168.12.1", "7"),
+            ("1.3.6.1.2.1.15.3.1.7.192.168.12.1", "192.168.12.1"),
+            ("1.3.6.1.2.1.15.3.1.14.192.168.12.1", '"06 04 "'),
+            ("1.3.6.1.2.1.15.3.1.2.192.168.12.1", "6"),
+        ]
+
+        # load trapd resulting file
+        # tgen.mininet_cli()
+
+        snmptrapfile = "{}/{}/snmptrapd.log".format(r2.logdir, r2.name)
+        outputfile = open(snmptrapfile).read()
+        output = snmp.trap(outputfile)
+        return output == expected
+
+    # skip tests is SNMP not installed
+    if not os.path.isfile("/usr/sbin/snmptrapd"):
+        error_msg = "SNMP not installed - skipping"
+        pytest.skip(error_msg)
+
+    snmptrapfile = "{}/{}/snmptrapd.log".format(r2.logdir, r2.name)
+    trap_file = open(snmptrapfile, "w")
+    trap_file.truncate(0)
+    trap_file.close()
+    topotest.sleep(1)
+    r1.vtysh_cmd("clear bgp *")
+    _, result = topotest.run_and_expect(_snmptrap_ipv4, True, count=2, wait=10)
+    assertmsg = "Can't fetch SNMP trap for ipv4"
+    assert result, assertmsg
+
+    def _snmptrap_ipv6():
+        expected = [
+            ("1.3.6.1.3.5.1.1.2.1.13.1.1.192.168.12.1", "7"),
+            ("1.3.6.1.3.5.1.1.2.1.6.1.1.192.168.12.1", "179"),
+            ("1.3.6.1.3.5.1.1.3.1.1.1.1.192.168.12.1", "6"),
+            ("1.3.6.1.3.5.1.1.3.1.2.1.1.192.168.12.1", "4"),
+            ("1.3.6.1.3.5.1.1.3.1.4.1.1.192.168.12.1", '"00 "'),
+            ("1.3.6.1.3.5.1.1.2.1.13.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "7"),
+            ("1.3.6.1.3.5.1.1.2.1.6.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "179"),
+            ("1.3.6.1.3.5.1.1.3.1.1.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "6"),
+            ("1.3.6.1.3.5.1.1.3.1.2.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "4"),
+            (
+                "1.3.6.1.3.5.1.1.3.1.4.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1",
+                '"00 "',
+            ),
+            ("1.3.6.1.3.5.1.1.2.1.13.1.1.192.168.12.1", "6"),
+            ("1.3.6.1.3.5.1.1.2.1.6.1.1.192.168.12.1", "179"),
+            ("1.3.6.1.3.5.1.1.2.1.13.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "6"),
+            ("1.3.6.1.3.5.1.1.2.1.6.1.2.32.1.13.184.0.0.0.0.0.0.0.0.0.18.0.1", "179"),
+        ]
+
+        # load trapd resulting file
+        # tgen.mininet_cli()
+
+        snmptrapfile = "{}/{}/snmptrapd.log".format(r2.logdir, r2.name)
+        outputfile = open(snmptrapfile).read()
+        output = snmp.trap(outputfile)
+        return output == expected
+
+    snmptrapfile = "{}/{}/snmptrapd.log".format(r2.logdir, r2.name)
+    trap_file = open(snmptrapfile, "w")
+    trap_file.truncate(0)
+    trap_file.close()
+    topotest.sleep(1)
+    r2.vtysh_cmd("conf\nbgp snmp traps bgp4-mibv2")
+    r2.vtysh_cmd("conf\nno bgp snmp traps rfc4273")
+    topotest.sleep(1)
+    r1.vtysh_cmd("clear bgp *")
+    topotest.sleep(1)
```

Comment:
Here too, this isn't a deterministic approach...

Reply:
taken into account